### PR TITLE
Bib availability end point

### LINKED ISSUE #1118: Add multiple bibs availability endpoint that gives bib-level availability only
This is needed in order to combine the data needed for the availability display on a search results page into a single API call.

It will call the Alma `bibs` endpoint and get the availability out of the AVA / AVE that come back.

Follow-up issue needed on the Orangelight side to use this endpoint.

proposed endpoint: `/bibliographic/availability?mmids=[id1,id2]`
returns:
```
{ mmsid: 991234,
  availability: ["available" | "unavailable" | "check_holdings"],
  availability_label: ["Available" | "Unavailable" | "Some items not available"]
}
```

### PR DESCRIPTION
**Pull request by hectorcorrea**
Adds two new endpoints to provide availability information for a single bib ID and for many bib IDs. 

```
/bibliographic/99122426947506421/availability.json
/bibliographic/availability.json?bib_ids=99122426947506421,9922486553506421
```

The response in these endpoints is meant to mimic what Orangelight expects in the search results page and details page. Notice that some properties as hard coded at this point (e.g. `on_reserve`) as we work out the differences between Voyager and Alma.

There are also a couple of new properties in the response (namely `holding_type` and `id`) that I am prototyping to handle new data that is now available in Alma but was not in Voyager in particular portfolios.

Fixes #1118 and #1145

### CODE REVIEW

File: app/adapters/alma_adapter.rb [hunk 1]
```diff
@@ -29,6 +29,24 @@ def get_availability(ids:)
...
+    return nil if bibs.count == 0
+    { bibs.first.id => AvailabilityStatus.new(bib: bibs.first).bib_availability }
+  end
+
+  # Returns availability for a list of bib ids
+  def get_availability_many(ids:)
+    bibs = Alma::Bib.find(Array.wrap(ids), expand: ["p_avail", "e_avail", "d_avail", "requests"].join(",")).each
+    return nil if bibs.count == 0
+    availability = {}
+    bibs.each do |bib|
```
Comment (eliotjordan) [inline app/adapters/alma_adapter.rb:44]: suggestion: Using `.each_with_object({})`  might be a good way to simplify this method.

File: app/adapters/alma_adapter/availability_status.rb [hunk 2]
```diff
@@ -10,6 +10,60 @@ def initialize(bib:)
       @bib = bib
     end
 
+    # Returns availability information for each of the holdings in the Bib record.
+    def bib_availability
+      availability = {}
+      holdings.each do |holding|
```
Comment (eliotjordan) [inline app/adapters/alma_adapter/availability_status.rb:16]: suggestion: same comment about using `.each_with_object({})`. Example on line 68.

File: app/adapters/alma_adapter/availability_status.rb [hunk 3]
```diff
@@ -10,6 +10,60 @@ def initialize(bib:)
...
+                   location: holding["location_code"],
+                   label: holding["location"],
+                   status_label: holding["availability"],
+                   more_items: (holding["total_items"] || "").to_i > 1,
+                   holding_type: "physical",
+                   id: holding["holding_id"]
+                 }
+               elsif holding["portfolio_pid"]
+                 # This kind of data is new with Alma.
+                 status = {
```
Comment (eliotjordan) [inline app/adapters/alma_adapter/availability_status.rb:42]: issue: we shouldn't need `status=` if we have already set `status=` on line 30.

File: app/adapters/alma_adapter/availability_status.rb [hunk 4]
```diff
@@ -10,6 +10,60 @@ def initialize(bib:)
...
+                   label: "N/A",
+                   status_label: holding["activation_status"],
+                   more_items: nil,
+                   holding_type: "portfolio",
+                   id: holding["portfolio_pid"]
+                 }
+               else
+                 # TODO: can there be more than one like this per bib?
+                 # If so we'll need to use unique IDs here.
+                 status = {
```
Comment (eliotjordan) [inline app/adapters/alma_adapter/availability_status.rb:54]: issue: same comment about `status=`.

**Review by eliotjordan — CHANGES_REQUESTED** [review]
This is looking good. I left a couple suggestions and an issue with redundant variable assignment.

Also, the bibliographic controller needs a few tests:
https://coveralls.io/jobs/77330258/source_files/7146501782#L26

As well as a test for a holding status block:
https://coveralls.io/jobs/77330258/source_files/7146501626#L51

File: app/adapters/alma_adapter.rb (hunk 1 above)
Comment (hectorcorrea) [reply app/adapters/alma_adapter.rb]: Done.

File: app/adapters/alma_adapter/availability_status.rb (hunk 2 above)
Comment (hectorcorrea) [reply app/adapters/alma_adapter/availability_status.rb]: Done.

File: app/adapters/alma_adapter/availability_status.rb (hunk 3 above)
Comment (hectorcorrea) [reply app/adapters/alma_adapter/availability_status.rb]: Good catch. Fixed.

File: app/adapters/alma_adapter/availability_status.rb (hunk 4 above)
Comment (hectorcorrea) [reply app/adapters/alma_adapter/availability_status.rb]: Ditto.

**Comment by hectorcorrea** [comment]
@eliotjordan Thanks for the review. I've made the changes that you suggested and added a few tests.

**Comment by hectorcorrea** [comment]
Replaced with https://github.com/pulibrary/bibdata/pull/1171

**Pull request closed without merging**